### PR TITLE
[back] feat: enable localized error messages in API

### PR DESCRIPTION
**Pull request by amatissart**
Related to #329 

Basically, this PR adds the `LocaleMiddleware` provided by Django, used for language detection:
https://docs.djangoproject.com/en/3.2/topics/i18n/translation/#how-django-discovers-language-preference

As DRF provides translated error messages for its validation errors, they will be present in API responses and displayed by frontend consistently with the current language, based on the standard `Accept-Language` that has been configured in #385.

A small side-effect of this change is about caching. `cache_page` uses the current language in its cache key. So a custom decorator `cache_page_no_i18n` has been implemented to force the default locale on API views where caching a distinct response per language would be wasteful.

### CODE REVIEW

File: backend/tournesol/utils/cache.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+from django.conf import settings
+from django.views.decorators.cache import cache_page
+from django.utils import translation
```
Comment (GresilleSiffle) [inline backend/tournesol/utils/cache.py:3]: We can swap these two import statements to keep them alphabetically ordered as recommended by `isort` ( which is recommended by Django ).

```suggestion
from django.utils import translation
from django.views.decorators.cache import cache_page
```

see: https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#imports

**Review by GresilleSiffle — APPROVED** [review]
:+1: 

Approved with small coding style suggestion.

File: backend/tournesol/utils/cache.py (hunk 1 above)
Comment (lfaucon) [reply backend/tournesol/utils/cache.py:3]: @GresilleSiffle Do you want to add isort to run automatically in Git-Hook / CI ? 

https://pycqa.github.io/isort/docs/configuration/git_hook.html

**Review by lfaucon — APPROVED** [review]

File: backend/tournesol/utils/cache.py (hunk 1 above)
Comment (GresilleSiffle) [reply backend/tournesol/utils/cache.py:3]: Yes please : )

If no objection I would like too. I'll just finish the other tasks I'm working on before.

Comment (amatissart) [reply backend/tournesol/utils/cache.py:3]: At first I thought that could be enforced using `pylint`, but it seems pylint only checks that standard library modules are imported before external ones.

Still, a wider issue could be about enforcing `pylint` checks, possibly only a subset of the current rules if they are too constraining. For now only a global score is computed, with a threshold at 8.5 / 10.

**Pull request merged**